src/graphql.ts

```diff
@@ -1,18 +1,21 @@
-type Operation = 'query' | 'mutation'
-
-const MUTATION_OP: Operation = 'mutation'
-
 /**
  * Returns the stringified JSON payload which corresponds to the GraphQL operation.
  * If no operation is explicitly defined, it will default to `query`.
  *
  * @param data - the raw HTTP request payload
  * @returns `{"query": data}` or `{"mutation": data}`
  */
-export function graphqlPayloadFor(data: string): string {
-  if (data.startsWith(MUTATION_OP)) {
-    return JSON.stringify({mutation: data})
-  }
-
-  return JSON.stringify({query: data})
+export function graphqlPayloadFor(
+  rawData: string,
+  rawVariables: string
+): string {
+  const data = minify(rawData)
+  const variables = minify(rawVariables)
+  return JSON.stringify({query: data, variables: JSON.parse(variables)})
```

Comment:
What about "mutation" ?

Reply:
The [GraphQL design spec](https://graphql.org/learn/serving-over-http/#post-request) states that both mutations and queries go in the `query` field.

I see that we are still missing a couple of things from the spec, such as:
- `operationName` attribute: used when we send multiple queries. It defined the query which is the point of entry for the overall request
- `application/graphql` header: if present, the request payload is not a JSON, but a GraphQL string

I added both enhancements to our [board](https://github.com/CamiloGarciaLaRotta/watermelon-http-client/projects/1#column-8582558)